src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmBrowserTestRunner.cs

```diff
@@ -164,7 +164,14 @@ private async Task RunConsoleMessagesPump(WebSocket socket, TaskCompletionSource
                         var line = Encoding.UTF8.GetString(mem.GetBuffer(), 0, (int)mem.Length);
                         line += Environment.NewLine;
 
-                        _messagesProcessor.Invoke(line);
+                        try
+                        {
+                            _messagesProcessor.Invoke(line);
+                        }
+                        catch (Exception ex) when (_messagesProcessor.WasmExitReceivedTcs.Task.IsCompletedSuccessfully)
+                        {
+                            _logger.LogWarning($"Test has return a result already, but message processor threw {ex} while logging message: {line}");
```

Comment:
Beware that if you do `threw {ex}`, then `ex.ToString()` returns all details with stack trace. So the message will be broken up.

```suggestion
                            _logger.LogWarning($"Test has returned a result already, but message processor threw {ex} while logging message: {line}");
```

Reply:
Yeah, I want the full trace here.

Reply:
Yeah, but do you want it in the middle of the sentence?

Maybe doing this is better?
```csharp
$"Test has returned a result already, but message processor threw {name of type only} while logging message: {line}{Environment.NewLine}{ex}"
```

Reply:
> Yeah, but do you want it in the middle of the sentence?

Heh I completely missed what you meant ;) /fixes